tests/unit/test_main.py

```diff
@@ -81,41 +81,37 @@ def load_plugins_mock(mocker, monkeypatch):
     return load_plugins_mock
 
 
-args = 'error_type'
-params = [
-    'list', 'obj'
-]
-
-
-@pytest.mark.parametrize(args, params)
-def test_log_or_exit_on_exceptions_no_debug(error_type, plugin_exc_mock,
-                                            mocker, monkeypatch):
+@pytest.mark.parametrize('error_type', ['list', 'obj'])
+def test_log_or_exit_on_exceptions_no_debug(
+        error_type, plugin_exc_mock, caplog, capsys):
     """Raise SystemExit if debug flag is off."""
-    logging_mock = mocker.MagicMock(main.logging, autospec=True)
-    monkeypatch.setattr(main, 'logging', logging_mock)
-
-    error = ('bad.plugin', plugin_exc_mock)
+    error = plugin_exc_mock
     if error_type == 'list':
-        error = [('bad.plugin', error)]
+        error = [error]
     with pytest.raises(SystemExit) as e:
         main._log_or_exit_on_exceptions('base msg', error, debug=False)
-
     e.match('1')
```

Comment:
Also not clear on where this '1' comes from...

Reply:
Shouldn't this be at least '(1)' or something then?  I think the exception message includes the entire trace, doesn't it?

Reply:
(Either way if it's not going to include more of the error message, I think this needs context-supplying comments.)

Reply:
There were some discussions earlier about how much to test error messages, and given that `e` is of type `SystemExit`, I think it is somewhat clear what it means, but I'm not opposed to adding more. It is outside the scope of this PR though.

The changes to this test in this PR are to pass the correct parameters to `_log_or_exit_on_exceptions` which expects an error, or list of errors, not a tuple of plugin name and error, or list of tuples.  Because this test was using a logging mock, it didn't matter that it was using the incorrect parameters, and we never caught the bug.  It only started to break when I replaced the mock with capsys to capture real logs.

Reply:
I do not understand why we would want to mock more things when testing against the real object works, using a standard testing fixture to capture output.

I don't know what you mean by "so...". Feel free to submit a PR to correct anything you think is incorrect.